packages/regenesis-surgery/test/predeploy.spec.ts

```diff
@@ -1,34 +1,178 @@
+
+          // check storage
+        })
+
+        it('should have the same nonce and balance as before', async () => {
+          const preNonce = await oldStateProvider.getTransactionCount(
+            account.address,
+            env.config.stateDumpHeight
+          )
+          const postNonce = await env.postL2Provider.getTransactionCount(
+            account.address
+          )
+          expect(preNonce).to.deep.eq(postNonce)
 
-    it('should have the same nonce and balance as before', async () => {})
+          const preBalance = await oldStateProvider.getBalance(
+            account.address,
+            env.config.stateDumpHeight
+          )
+          const postBalance = await env.postL2Provider.getBalance(
+            account.address
+          )
+          expect(preBalance).to.deep.eq(postBalance)
+        })
+      })
+    }
   })
 
   describe('predeploys where the old state should be preserved', () => {
-    it('should have the code of the base genesis file', async () => {})
+    for (const [i, account] of predeploys.noWipe.entries()) {
+      describe(`account ${i}/${predeploys.noWipe.length} (${account.address})`, () => {
+        it('should have the code of the base genesis file', async () => {
+          const preBytecode = await genesisStateProvider.getCode(
+            account.address
+          )
+          const postBytecode = await env.postL2Provider.getCode(account.address)
+          expect(preBytecode).to.eq(postBytecode)
+        })
 
-    it('should have the combined storage of the old and new state', async () => {})
+        it('should have the combined storage of the old and new state', async () => {
+          // check storage
+        })
 
-    it('should have the same nonce and balance as before', async () => {})
+        it('should have the same nonce and balance as before', async () => {
+          const preNonce = await oldStateProvider.getTransactionCount(
+            account.address,
+            env.config.stateDumpHeight
+          )
+          const postNonce = await env.postL2Provider.getTransactionCount(
+            account.address
+          )
+          expect(preNonce).to.deep.eq(postNonce)
+
+          const preBalance = await oldStateProvider.getBalance(
+            account.address,
+            env.config.stateDumpHeight
+          )
+          const postBalance = await env.postL2Provider.getBalance(
+            account.address
+          )
+          expect(preBalance).to.deep.eq(postBalance)
+        })
+      })
+    }
   })
 
   describe('OVM_ETH', () => {
-    it('should have disabled ERC20 features', async () => {})
-
-    it('should no recorded balance for the contracts that move to WETH9', async () => {})
+    for (const [i, account] of predeploys.eth.entries()) {
+      describe(`account ${i}/${predeploys.eth.length} (${account.address})`, () => {
+        it('should have disabled ERC20 features', async () => {
+          // .transfer should not work? need live provider
```

Comment:
i switched some of the ETH / WETH tests based on where makes most sense, lmk if my understanding is correct! going to start adding some of these live provider tests